documentation/course/e2/README.md

```diff
@@ -1646,10 +1646,95 @@ Z uwagi na dużą ilość danych, zdecydowano się na czas retencji kopii zapaso
+    <td>b-tree</td>
+    <td>Indeks wspierający wyszukiwanie mandatów według inspektora.</td>
+  </tr>
+  <tr>
+    <td><code>wallet.passenger_id</code></td>
+    <td>b-tree</td>
+    <td>Indeks wspierający wyszukiwanie portfeli pasażerów.</td>
+  </tr>
+  <tr>
+    <td><code>credit_card_info.wallet_id</code></td>
+    <td>b-tree</td>
+    <td>Indeks wspierający wyszukiwanie kart kredytowych według portfela.</td>
+  </tr>
+  <tr>
+    <th colspan="3">Ograniczenia</th>
+  </tr>
+  <tr>
+    <td colspan="3"><code>wallet.balance_pln >= 0</code></td>
+  </tr>
+  <tr>
+    <td colspan="3"><code>wallet.passenger_id UNIQUE</code></td>
+  </tr>
+  <tr>
+    <td colspan="3"><code>credit_card_info.holder_name <> ''</code></td>
+  </tr>
+  <tr>
+    <td colspan="3"><code>fine.recipient <> ''</code></td>
+  </tr>
+  <tr>
+    <td colspan="3"><code>fine.amount_pln > 0</code></td>
+  </tr>
+  <tr>
+    <td colspan="3"><code>fine.time <= NOW()</code></td>
+  </tr>
+</table>
+
+Z uwagi na średnie obciążenie bazy danych w podsystemie `inferius`, jako klasę instancji wybrano `db.m7g.large`. Wersja `large` oferuje 2 vCPU oraz 8 GiB RAM. Baza przechowuje istotne i wrażliwe dane, zatem kluczowe jest włączenie szyfrowania.
+
+Jako górną estymację fizycznego rozmiaru wiersza bazy danych przyjęto sumę maksymalnych rozmiarów wszystkich kolumn z pominięciem dodatkowej pamięci wykorzystywanej przez bazę danych do reprezentacji struktur danych, daje to następujące rozmiary wierszy dla tabel:
+
+- `wallet`: 16 + 16 + 8 = 40 bajtów,
+- `credit_card_info`: 16 + 16 + 255 + 16 + 255 + 4 = 562 bajty,
+- `fine`: 16 + 16 + 16 + 255 + 8 + 8 + 32 + 32 = 383 bajty.
+
+Dodatkowo indeksy na tabeli mają następujące estymowane rozmiary na każdy wiersz:
+
+- `wallet`: 24 + 24 = 48 bajtów,
+- `credit_card_info`: 24 + 24 = 48 bajtów,
+- `fine`: 24 + 24 + 24 = 72 bajty.
+
+Zakładając, że we Wrocławiu mieszka 825 tys. osób[^ludnosc-wroclawia] oraz odwiedza go 1.2 mln turystów rocznie[^turysci-wroclawia] oraz że każda osoba bedzie posiadała swoją aplikację to górna granica wynosi **2 mln unikalnych użytkowników** (2 mln * 88 = 176 MB) w pierwszym roku działania systemu oraz **wzrost o maksymalnie 1.2 mln kont rocznie** (105 MB). Zakładając że każda osoba doda do swojego konta 2 karty kredytowe, to w pierwszym roku działania systemu będzie to 2 mln * 610 = 1.22 GB, a rocznie 1.2 mln * 610 = 732 MB. Dodatkowo zakładając, że **rocznie kontrolerzy wystawiają 45 000**[^roczne-mandaty] mandatów, to roczny przyrost danych wynosi około 45 000 * 455 = 20 MB. Sumarycznie, roczny przyrost danych wynosi około 2 GB rocznie. Ze względu na to, że większość aproksymacji wykonanych tutaj zawyżają wynik, zdecydowano się na początkowy rozmiar bazy danych na RDS wynoszący **20** GB.
+
+Zdecydowano się na czas retencji kopii zapasowych wynoszący 7 dni, co powinno dać wystarczająco dużo czasu na zauważenie i naprawienie błędów, a jednocześnie nie przechowuje danych zbyt długo.
```

Comment:
![obraz](https://github.com/user-attachments/assets/ac9ce44e-61a5-4374-b3e2-8ed940779311)

**issue:** I don't think we should hold backup for 5 years, but I'd also say that it should be longer than one week.

Whereas account data should be kept as short as possible (GDPR/RODO, right to be forgotten, etc.), financial and transactional data should be kept as long as possible.

Maximum backup window supported by RDS lasts 35 days. I'd probably choose 1 month if I were you.